src/udev/udev-builtin-blkid.c

```diff
@@ -234,6 +240,21 @@ static int builtin_blkid(sd_device *dev, int argc, char *argv[], bool test) {
                         break;
 
                 switch (option) {
+                case 'H':
+#if HAVE_BLKID_PROBE_SET_HINT
+                        errno = 0;
+                        r = blkid_probe_set_hint(pr, optarg, 0);
+                        if (r < 0)
+                                return log_device_error_errno(dev, errno > 0 ? errno : ENOMEM, "Failed to use '%s' probing hint: %m", optarg);
+                        break;
+#else
+                        /* Use the hint <name>=<offset> as probing offset for old versions */
+                        optarg = strchr(optarg, '=');
+                        if (!optarg)
+                                return log_device_error_errno(dev, SYNTHETIC_ERRNO(ERANGE), "Failed to parse probing hint: %m");
```

Comment:
I think EINVAL is better in this case. Also, it is better to include the original `optarg` on error:
```
{
        char equal = strchr(optarg, '=');
        if (!equal)
                return log_device_error_errno(dev, SYNTHETIC_ERRNO(EINVAL), "Invalid syntax for --hint option: %s", optarg);

        optarg = equal + 1;
}
_fallthrough_;
```

Reply:
After reading https://github.com/karelzak/util-linux/commit/67719fbbda6de60c63496e62dae8617336f9ac7b, and https://github.com/karelzak/util-linux/commit/ac3a0fd92e571ccc46919bc0a651d0647e7b5d36, `blkid` supports `--hint` without value, and in that case, `0` is used for the offset. Maybe, we should also follow that way?

Reply:
Good point. Now we have no use for hints without value, but maybe one day it will be a way how to inform libblkid about some detection quirks.

For the fallback, it means no error message, and no-op as zero is the default probing offset:
```C
                        /* Use the hint <name>=<offset> as probing offset for old versions */
                        optarg = strchr(optarg, '=');
                        if (!optarg)              
                                /* no value means 0, do nothing for old versions */
                                break;            
                        optarg++;
                        _fallthrough_;    
 #endif                          
                case 'o':       
                        r = safe_atoi64(optarg, &offset);
```

Reply:
Hint `session_offset` does not make sense without value. And this hint is the only used hint. Does it make sense to support hints without value til beginning?

Reply:
I think the --hint should be implemented as a generic option and no specifically to `session_offset`. Anyway, it's an implementation detail.